src/Monolog/Formatter/JsonFormatter.php

```diff
@@ -216,7 +216,7 @@ protected function normalize(mixed $data, int $depth = 0): mixed
      * Normalizes given exception with or without its own stack trace based on
      * `includeStacktraces` property.
      *
-     * @return array<string, string|int|array<string|int|array<string>>>|string
+     * @return array<array-key, string|int|array<string|int|array<string>>>
```

Comment:
keys might be non-string when max depth exceeds, and maybe in case of JsonSerializable